components/03-modules/modal/Modal.js

```diff
@@ -69,36 +67,55 @@ export default {
       this.ariaHidden = state ? 'false' : 'true'
       this.visibility = state
 
-      if (state === true) {
-        this.$nextTick(() => this.$refs.modal.focus())
-      } else {
+      if(state) {
         this.$nextTick(() => this.focused.focus())
+      } else {
+        this.$nextTick(() => this.trigger.focus())
```

Comment:
please check if we can get rid of this condition

Reply:
I removed unnecessary data "focused" and above statement. I've moved the appropriate focus setting to show and hide methods.